### PR TITLE
Push Tasks Shared Context 

### LINKED ISSUE #26508: CLI  Files Push - continues to attempt folder removal after having succeed the first time
### Parent Issue

https://github.com/dotCMS/core/issues/26380

### Problem Statement

Our "CLI Files Push" command explores the local file system and makes a comparison against the remote contents of the dotCMS instance it wants to synch with. 
Once the command recognizes a folder that needs to be removed it creates a tree-like data structure that represents all the files and folders hierarchically organized. Then that tree-like structure gets passed down to the next service that will process all the items marked for deletion.
The problem is that once a folder is deleted for a language or state combination.  
subsequent attempts to delete that same folder under other language and state combinations will error out with an ugly exception. 

### Steps to Reproduce

When testing the solution for https://github.com/dotCMS/core/issues/26380 
You should see 404 error messages for each and every combination of status and language where the folder could have been 

### Acceptance Criteria

Once a folder is marked for delete. Only one single attempt should be made. All unnecessary subsequent attempts should be dropped deletion

### dotCMS Version

current master

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
Refs: #26508

### Proposed Changes
* Our CLI explores the file structure branches on the workspace using Tasks 
* Every Task is a Thread 
* On This PR We're introducing a context that basically can be used to share info across TraverseTask
* For now We use the context as a key holder to store the info of the asset that has been pushed, deleted, or archived
* This avoids carrying out repetitive operations that were generating occasionally a 404 
* The context gets cleared when the "retry" flag is passed.
* Additionally to that, some additional refactoring that introduces a Builder Pattern simplifies how we pass params

### Summary
🚀🛠️🧹

This pull request refactors the dotCMS CLI file push operation to use new classes that simplify the parameter passing and the traversal logic. It introduces the `LocalTraverseParams` and `PushTraverseParams` classes for local and remote traversal options, and the `AbstractPushTraverseParams` interface for common push parameters. It also removes unused or unnecessary parameters, imports, and logic from the `PushService`, `PushServiceImpl`, `RemoteTraversalService`, and `RemoteTraversalServiceImpl` classes.

> _We are the pushers of the code_
> _We traverse the files with our mode_
> _We simplify the parameters of doom_
> _We refactor the classes with our broom_

### Walkthrough
*  Add `AbstractPushTraverseParams` interface to define common parameters for push traversal operation (F3)
*  Replace `TraverseParams` class with `LocalTraverseParams` class that extends `AbstractLocalTraverseParams` interface and provides builder for local traversal parameters ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L80-R80), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-b6344d4aea8d7792f1789e2402e8acfe71d8c8924916b031c8ef13eded171193L23-R23), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-90c9d8e0abf28d7d80546ba32d273bf023a012dc1a452e47e7ae46d18699e826L52-R52), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-90c9d8e0abf28d7d80546ba32d273bf023a012dc1a452e47e7ae46d18699e826L86-R86))
*  Change `processTreeNodes` and `pushTreeNode` methods to use `PushTraverseParams` class instead of multiple arguments, and remove unused parameters ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-27eebe367a8b2d56004991d0774c988d35b092459700be7f0d3ae3d9db20feadL36-R45), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L188-R199), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L230-R230), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-508b13e62294e3bc4ab3b2fd56556d675e4df7807452dea67e37b5a210de7491L43-R47), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-02c6eb1a767df6cea255b6db297dcdf324b0d3abdcb7e99d96f2fc38eb251bc3L110-R135))
*  Add `params.pushContext().clear()` statement to clear push context before retrying push operation ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L214-R209))
*  Move `retryAttempts` variable to `else` block and replace `maxRetryAttempts` variable with `params.maxRetryAttempts()` method ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L214-R209), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L272-R271), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L279-R278))
*  Fix typo in comment ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L129-R129))
*  Remove unused imports ([link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-85e8e28e01775839f3e70dc752471422cadc81375a3682941fc28a2f4f2a2b77L16), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-508b13e62294e3bc4ab3b2fd56556d675e4df7807452dea67e37b5a210de7491L4-L5), [link](https://github.com/dotCMS/core/pull/26543/files?diff=unified&w=0#diff-02c6eb1a767df6cea255b6db297dcdf324b0d3abdcb7e99d96f2fc38eb251bc3L9-R13))

### CODE REVIEW

**Review by jgambarios — APPROVED** [review]

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/PushService.java [hunk 1]
```diff
@@ -33,21 +31,17 @@ List<TraverseResult> traverseLocalFolders(
...
-     * @param workspace          the workspace path
-     * @param localPathStructure the local path structure of the folder being pushed
-     * @param treeNode           the tree node representing the folder and its contents with all the push information
-     *                           for each file and folder
-     * @param treeNodePushInfo   the push information summary associated with the tree node
-     * @param failFast           true to fail fast, false to continue on error
-     * @param maxRetryAttempts   the maximum number of retry attempts in case of error
+     * @param output           the output option mixin
+     * @param maxRetryAttempts the maximum number of retry attempts in case of error
+     * @param treeNodePushInfo
```
Comment (nollymar) [inline tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/PushService.java:40]: can you please add the doc for the `treeNodePushInfo` and `params`? Part of the existing doc was removed (`treeNodePushInfo`). I would even rename the `params` param

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/PushServiceImpl.java [hunk 2]
```diff
@@ -185,50 +185,49 @@ boolean isAllFoldersMarkedForDelete(List<TreeNode> nodes) {
...
-     * @param output             the output option mixin
-     * @param workspace          the workspace path
-     * @param localPathStructure the local path structure of the folder being pushed
-     * @param treeNode           the tree node representing the folder and its contents with all the push information
-     *                           for each file and folder
-     * @param treeNodePushInfo   the push information summary associated with the tree node
-     * @param failFast           true to fail fast, false to continue on error
-     * @param maxRetryAttempts   the maximum number of retry attempts in case of error
+     * @param output   the output option mixin
+     * @param pushInfo
```
Comment (nollymar) [inline tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/PushServiceImpl.java:193]: Please update the doc for `pushInfo` and `params` too

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/traversal/RemoteTraversalService.java [hunk 3]
```diff
@@ -40,20 +38,12 @@ Pair<List<Exception>, TreeNode> traverseRemoteFolder(
...
      *
-     * @param workspace          the local workspace path
-     * @param localPathStructure the local path structure of the folder being pushed
-     * @param treeNode           the tree node representing the folder and its contents with all the push
-     *                           information for each file and folder
-     * @param failFast           true to fail fast, false to continue on error
-     * @param isRetry            true if this is a retry attempt, false otherwise
-     * @param progressBar        the console progress bar to track and display the push progress
-     * @return A list of exceptions encountered during the push process.
+     * @param params@return A list of exceptions encountered during the push process.
```
Comment (nollymar) [inline tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/traversal/RemoteTraversalService.java:45]: same here

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/traversal/task/PushTreeNodeTask.java [hunk 4]
```diff
@@ -24,64 +28,33 @@
...
-     * @param workspacePath      the local workspace path
-     * @param localPathStructure the local path structure of the folder being pushed
-     * @param rootNode           the tree node representing the folder and its contents with all the push
-     *                           information for each file and folder
-     * @param failFast           true to fail fast, false to continue on error
-     * @param isRetry            true if this is a retry, false otherwise
-     * @param logger             the logger for logging debug information
-     * @param pusher             the pusher for pushing assets and folders
-     * @param progressBar        the console progress bar to track and display the push progress
+     * @param params
```
Comment (nollymar) [inline tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/files/traversal/task/PushTreeNodeTask.java:38]: same here

**Review by nollymar — APPROVED** [review]

**Pull request merged**